txp/Pos/Txp/MemState/Class.hs

```diff
@@ -41,73 +41,62 @@ type MonadTxpMem ext ctx m
 askTxpMem :: MonadTxpMem ext ctx m => m (GenericTxpLocalData ext)
 askTxpMem = view (lensOf @TxpHolderTag)
 
-getTxpLocalData
+-- | Operate with some or all of the TXP local data.
+--
+--   Since this function takes an STM action, it can be used to
+--   read or modify the components.
+withTxpLocalData
     :: (MonadIO m, MonadTxpMem e ctx m)
     => (GenericTxpLocalData e -> STM.STM a) -> m a
-getTxpLocalData getter = askTxpMem >>= \ld -> atomically (getter ld)
+withTxpLocalData f = askTxpMem >>= \ld -> atomically (f ld)
```

Comment:
Ah, much nicer now. `getTxpLocalData` was suggesting we were just reading stuff, but indeed as we can do any kind of modifications, this is a much more effective name.